broker/ga4gh/broker/endpoints/repositories.py

```diff
@@ -0,0 +1,287 @@
+        id_charset = eval(id_charset)
+    except Exception:
+        id_charset = ''.join(sorted(set(id_charset)))
+
+    for i in range(retries + 1):
+        logger.debug(f"Trying to insert/update object: try {i}")
+
+        repo_id = generate_id(
+            charset=id_charset,
+            length=id_length,
+        )
+        access_token = str(generate_id(id_charset, 32))
+        repository_object['id'] = repo_id
+        repository_object['access_token'] = access_token
+        try:
+            db_collection.insert_one(repository_object)
+            break
+        except DuplicateKeyError:
+            logger.error(f"DuplicateKeyError ({repo_id}): Key generated"
+                         f" is already present.")
+            continue
+    else:
+        logger.error(
+                    f"Could not generate unique identifier."
+                    f" Tried {retries + 1} times."
+                )
+        raise InternalServerError
+    if repository_object is not None and '_id' in repository_object:
+        del repository_object['_id']
+        del repository_object['url']
+    logger.info(f"Added object with '{repository_object}'.")
+    return repository_object
+
+
+def get_repositories():
+    """Get available Repositories.
+
+    Returns:
+        list of repository_objects
+
+    Description:
+        - Fetches all repositories from the mongodb.
+        - Removes subscription_list if present.
+        - Returns list of repository_objects.
+
+    Raises:
+        RepositoryNotFound: Raised when there is no registered repository in
+        the broker.
+
+    """
+    db_collection = (
+        current_app.config['FOCA'].db.dbs['brokerStore'].
+        collections['repositories'].client
+    )
+
+    cursor = db_collection.find(
+        {}, {'access_token': False, '_id': False}
+    )
+    repository_object = list(cursor)
+    for repo in repository_object:
+        if 'subscription_list' in repo:
+            del repo['subscription_list']
+    return list(repository_object)
+
+
+def generate_id(
+        charset: str = ''.join([string.ascii_lowercase, string.digits]),
+        length: int = 6
+) -> str:
+    """Generate random string based on allowed set of characters.
+
+    Args:
+        charset: String of allowed characters.
+        length: Length of returned string.
+
+    Returns:
+        Random string of specified length and composed of defined set of
+        allowed characters except '_' and '.' or '-' at index 0 and length-1
+    """
+    generated_string = ''
+    counter = 0
+    while counter < length:
+        random_char = choice(charset)
+        if random_char in '_':
+            continue
+        if counter == 0 or counter == (length - 1):
+            if random_char in '.-' or random_char in string.digits:
+                continue
+        counter = counter + 1
+        generated_string = generated_string + ''.join(random_char)
+    return generated_string
+
+
+def get_repository_info(repo_id: str):
```

Comment:
Better to use `get_repository()` as opposed to `get_repositories()` as this reflects the typical REST philosophy better